php/libraries/LorisForm.class.inc

```diff
@@ -677,45 +677,55 @@ class LorisForm
     protected function selectHTML($el)
     {
         $strOptions    = '';
-        $options       = $el['options'];
+        $options       = $el['options'] ?? array();
         $cls           = '';
         $changehandler = '';
         $multiple      = '';
         $id            = '';
         $disabled      = '';
-        $name          = $el['name'];
+        $name          = $el['name'] ?? '';
 
-        foreach ($options as $optionKey => $optionVal) {
-            $selected = '';
+        if (is_array($options) || $options instanceof Traversable) {
```

Comment:
Where do we have select elements being added with no options set? That seems like it should be an exception, not something that's supported. I can't think of any reasons that would be a valid use case.

This logic should also be inverted. Check for an invalid state, and immediately throw an exception (or return an error), then handle the happy path (which doesn't need any changes), rather than having the error message ~50 lines away from the check that triggered it..